pkg/cmd/cli/cmd/deploy.go

```diff
@@ -156,7 +156,8 @@ func (o *DeployOptions) Complete(f *clientcmd.Factory, args []string, out io.Wri
 
 func (o DeployOptions) Validate() error {
 	if len(o.deploymentConfigName) == 0 {
-		return errors.New("a deployment config name is required.")
+		msg := fmt.Sprintf("a deployment config name is required. Use \"%s get dc\" for a list of available deployment configs.", o.baseCommandName)
```

Comment:
Please add a "\n" between "required." and " Use" just like how the `oc start-build` does. 
Others LGTM. Thanks!